tests/e2e/performance/max_vector_dim_test.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/file"
+	"github.com/vdaas/vald/internal/log"
+	"github.com/vdaas/vald/internal/log/logger"
+	"github.com/vdaas/vald/internal/net/grpc/codes"
+	"github.com/vdaas/vald/internal/net/grpc/status"
+	"github.com/vdaas/vald/internal/test/data/vector"
+	"github.com/vdaas/vald/tests/e2e/kubernetes/client"
+	"github.com/vdaas/vald/tests/e2e/kubernetes/portforward"
+	"google.golang.org/genproto/googleapis/rpc/code"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+)
+
+const (
+	maxBit = 32
+	id     = "1"
+)
+
+var (
+	host                string
+	port                int
+	namespace           string
+	bit                 int
+	dim                 int
+	kubeClient          client.Client
+	forwarder           *portforward.Portforward
+	indexingWaitSeconds uint
+	kubeConfig          string
+	pf                  bool
+	pfPodName           string
+	pfPodPort           int
+	fileName            string
+)
+
+func init() {
+	testing.Init()
+
+	flag.StringVar(&host, "host", "localhost", "hostname")
+	flag.IntVar(&port, "port", 8081, "gRPC port")
+	flag.StringVar(&namespace, "namespace", "default", "namespace")
+	flag.IntVar(&bit, "bit", 1, "bit")
+	flag.UintVar(&indexingWaitSeconds, "wait", 30, "indexing wait seconds")
+	flag.StringVar(&fileName, "file", "tmp.log", "output file name")
+
+	flag.BoolVar(&pf, "portforward", false, "enable port forwarding")
+	flag.StringVar(&pfPodName, "portforward-pod-name", "vald-lb-gateway", "pod name (only for port forward)")
+	flag.IntVar(&pfPodPort, "portforward-pod-port", port, "pod gRPC port (only for port forward)")
+
+	flag.StringVar(&kubeConfig, "kubeconfig", file.Join(os.Getenv("HOME"), ".kube", "config"), "kubeconfig path")
+
+	flag.Parse()
+}
+
+func TestMain(m *testing.M) {
+	log.Init(log.WithLoggerType(logger.NOP.String()))
+	var err error
+	if pf {
+		kubeClient, err = client.New(kubeConfig)
+		if err != nil {
+			os.WriteFile(fileName, []byte(err.Error()), os.ModePerm)
+			os.Exit(1)
+		}
+		forwarder = kubeClient.Portforward(namespace, pfPodName, port, pfPodPort)
+		err = forwarder.Start()
+		if err != nil {
+			os.WriteFile(fileName, []byte(err.Error()), os.ModePerm)
+			os.Exit(1)
+		}
+	}
+
+	if bit < 1 || maxBit < bit {
+		err = errors.New("Invalid argument: bit should be 0 ~ 32. set bit was " + strconv.Itoa(bit))
+		os.WriteFile(fileName, []byte(err.Error()), os.ModePerm)
+		os.Exit(1)
+	}
+	_ = m.Run()
+	if forwarder != nil {
+		forwarder.Close()
+	}
+	os.Exit(0)
+}
+
+func TestE2EInsertOnlyWithOneVectorAndSearch(t *testing.T) {
+	t.Helper()
+	dim := 1 << bit
+	if bit == maxBit {
+		dim--
+	}
+	if dim > algorithm.MaximumVectorDimensionSize {
+		t.Fatalf("Invalid argument: dimension should be equal or under than " + strconv.Itoa(algorithm.MaximumVectorDimensionSize) + ". set dim was " + strconv.Itoa(dim))
+	}
+	ctx := context.Background()
+	conn, err := grpc.DialContext(
+		ctx,
+		host+":"+strconv.Itoa(port),
```

Comment:
Is it better to use `JoinHostPort` ? 🤔 

```suggestion
		net.JoinHostPort("localhost", strconv.Itoa(port)),
```

Reply:
```suggestion
		net.JoinHostPort(host, strconv.Itoa(port)),
```
This?

Reply:
Yes